libminifi/include/core/TypedValues.h

```diff
@@ -149,10 +144,36 @@ class DataSizeValue : public TransformableValue, public state::response::UInt64V
   }
 };
 
-}  // namespace core
-}  // namespace minifi
-}  // namespace nifi
-}  // namespace apache
-}  // namespace org
+class DataTransferSpeedValue : public TransformableValue, public state::response::UInt64Value {
+ public:
+  MINIFIAPI static const std::type_index type_id;
+
+  explicit DataTransferSpeedValue(const std::string &transfer_speed_string)
+      : state::response::UInt64Value(0) {
+    DataSizeValue::StringToInt<uint64_t>(removePerSecSuffix(transfer_speed_string), value);
+    string_value = transfer_speed_string;
+  }
+
+  explicit DataTransferSpeedValue(uint64_t value)
+      : state::response::UInt64Value(value) {
+  }
+
+  DataTransferSpeedValue()
+      : state::response::UInt64Value(0) {
+  }
+
+  static std::string removePerSecSuffix(const std::string &input) {
+    auto lower_case_input = utils::StringUtils::trim(utils::StringUtils::toLower(input));
+    if (lower_case_input.ends_with("/s")) {
+      return lower_case_input.substr(0, lower_case_input.size() - 2);
+    }
+    throw utils::internal::ParseException("Couldn't parse DataTransferSpeedValue, no valid suffix (/s, /sec, ps) found!");
```

Comment:
this error message is out of date

Reply:
Updated in 41364556e36680ca7f8d564ffa08fe110188a34c